### PR TITLE
chore(derived_code_mappings): Minor refactor

### PR DESCRIPTION
**Pull request by armenzg**
It makes writing the tests a little bit easier.

I separated these changes from #45088 to ease reviews.

### CODE REVIEW

File: src/sentry/integrations/github/client.py [hunk 1]
```diff
@@ -194,44 +194,43 @@ def _populate_repositories(self, gh_org: str, cache_seconds: int) -> List[Dict[s
 
         return repositories
 
+    def _populate_trees_process_error(self, error: ApiError, extra: Dict[str, str]) -> None:
```
Comment (armenzg) [inline src/sentry/integrations/github/client.py:197]: I just pulled this function from within `_populate_trees` so the function is easier to read.

image

File: tests/sentry/integrations/github/test_integration.py [hunk 2]
```diff
@@ -16,7 +16,7 @@
 from sentry.utils.cache import cache
 
 TREE_RESPONSES = {
-    "xyz": {"status_code": 200, "body": {"tree": [{"path": "src/foo.py", "type": "blob"}]}},
+    "xyz": {"status_code": 200, "body": {"tree": [{"path": "src/xyz.py", "type": "blob"}]}},
```
Comment (armenzg) [inline tests/sentry/integrations/github/test_integration.py:19]: It's odd that I have a repo `foo` but the `foo` file belongs to the `xyz` repo.

File: tests/sentry/integrations/github/test_integration.py [hunk 3]
```diff
@@ -585,7 +586,7 @@ def test_github_prevent_install_until_pending_deletion_is_complete(self):
             organization=self.organization, integration=integration
         ).exists()
 
-    def set_rate_limit(self, remaining, limit=5000):
+    def set_rate_limit(self, remaining=MINIMUM_REQUESTS + 100, limit=5000):
```
Comment (armenzg) [inline tests/sentry/integrations/github/test_integration.py:589]: Now calling `set_rate_limit` starts with a high enough value and makes calling the function easier to think about it.

File: tests/sentry/integrations/github/test_integration.py [hunk 4]
```diff
@@ -605,34 +606,41 @@ def get_installation_helper(self):
         installation = integration.get_installation(self.organization.id)
         return installation
 
+    def _expected_trees(self, repo_info_list=None):
```
Comment (armenzg) [inline tests/sentry/integrations/github/test_integration.py:609]: Helper functions. It makes writing the tests a little cleaner.

File: tests/sentry/integrations/github/test_integration.py [hunk 5]
```diff
@@ -605,34 +606,41 @@ def get_installation_helper(self):
...
-            "Test-Organization/foo": RepoTree(
-                Repo("Test-Organization/foo", "master"), ["src/sentry/api/endpoints/auth_login.py"]
-            ),
-            "Test-Organization/xyz": RepoTree(
-                Repo("Test-Organization/xyz", "master"), ["src/foo.py"]
-            ),
-        }
-
+        self.set_rate_limit()
+        expected_trees = self._expected_trees()
```
Comment (armenzg) [inline tests/sentry/integrations/github/test_integration.py:634]: 2 lines instead of 10.

**Review by scttcper — APPROVED** [review]

**Pull request merged**